compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2149,9 +2151,14 @@ impl<'tcx> Debug for Rvalue<'tcx> {
             UnaryOp(ref op, ref a) => write!(fmt, "{:?}({:?})", op, a),
             Discriminant(ref place) => write!(fmt, "discriminant({:?})", place),
             NullaryOp(ref op, ref t) => write!(fmt, "{:?}({:?})", op, t),
-            ThreadLocalRef(did) => ty::tls::with(|tcx| {
-                let muta = tcx.static_mutability(did).unwrap().prefix_str();
-                write!(fmt, "&/*tls*/ {}{}", muta, tcx.def_path_str(did))
+            ThreadLocalRef(did, ty) => ty::tls::with(|tcx| {
+                let muta = ty.builtin_deref(true).unwrap().mutbl.prefix_str();
+                let raw = match ty.kind() {
+                    ty::Ref(..) => "",
+                    ty::RawPtr(..) => "raw ",
```

Comment:
If you are matching anyway, the same match could also extract the mutability... do you think taht would make the code cleaner?